clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go

```diff
@@ -83,24 +85,39 @@ func (*waitForControlledPodsRunningMeasurement) Execute(config *measurement.Meas
 	if err != nil {
 		return summaries, err
 	}
-	runningRuntimeObjects := 0
-	for start := time.Now(); time.Since(start) < timeout; time.Sleep(defaultWaitForRCPodsInterval) {
-		if runningRuntimeObjects >= len(runtimeObjectsList) {
-			return summaries, nil
-		}
+	var wg wait.Group
+	errList := util.NewErrorList()
+	var runningRuntimeObjects int32
+	for objectCounter := 0; objectCounter < len(runtimeObjectsList); objectCounter++ {
+		objectIndex := objectCounter
+		wg.Start(func() {
+			if err := waitForRuntimeObject(config.ClientSet, runtimeObjectsList[objectIndex], timeout); err != nil {
+				errList.Append(fmt.Errorf("waiting for %v error: %v", kind, err))
+				return
+			}
 
-		if err := waitForRuntimeObject(config.ClientSet, runtimeObjectsList[runningRuntimeObjects], timeout-time.Since(start)); err != nil {
-			return summaries, fmt.Errorf("waiting for %v error: %v", kind, err)
-		}
-		runningRuntimeObjects++
-		if namespace == metav1.NamespaceAll {
-			glog.Infof("%s: %d / %d %ss with all pods running", waitForControlledPodsRunningName, runningRuntimeObjects, len(runtimeObjectsList), kind)
-		} else {
-			glog.Infof("%s: %v: %d / %d %ss with all pods running", waitForControlledPodsRunningName, namespace, runningRuntimeObjects, len(runtimeObjectsList), kind)
-		}
+			atomic.AddInt32(&runningRuntimeObjects, 1)
+			objName, err := getNameFromRuntimeObject(runtimeObjectsList[objectIndex])
+			if err != nil {
+				errList.Append(fmt.Errorf("reading object name error: %v", err))
+				return
+			}
+			objNamespace, err := getNamespaceFromRuntimeObject(runtimeObjectsList[objectIndex])
+			if err != nil {
+				errList.Append(fmt.Errorf("reading object namespace error: %v", err))
+				return
+			}
+			glog.Infof("%s: %s (%s) has all pods running", waitForControlledPodsRunningName, objName, objNamespace)
+		})
 	}
+	wg.Wait()
 
-	return summaries, fmt.Errorf("timeout while waiting for %ss to be running in namespace '%v' with labels '%v' and fields '%v' - only %d found running with all pods", kind, namespace, labelSelector, fieldSelector, runningRuntimeObjects)
+	if int(runningRuntimeObjects) == len(runtimeObjectsList) {
+		glog.Infof("%s: %d / %d %ss are running with all pods", waitForControlledPodsRunningName, runningRuntimeObjects, runningRuntimeObjects, kind)
```

Comment:
Something seems wrong with this log. Why are you printing `runningRuntimeObjects` twice?

Reply:
I've changed the second one to len(runtimeObjectsList).